modules/utxo-lib/test/bitgo/psbt/psbtUtil.ts

```diff
@@ -0,0 +1,278 @@
+    psbtParsed?.publicKeys.forEach((publicKey) => {
+      assert.deepStrictEqual(Buffer.isBuffer(publicKey), true);
+    });
+  } else {
+    assert.deepStrictEqual(txParsed.publicKeys.length, psbtParsed?.publicKeys?.length);
+    const pubKeyMatch = txParsed.publicKeys.every((txPubKey) =>
+      psbtParsed?.publicKeys?.some((psbtPubKey) => psbtPubKey.equals(txPubKey))
+    );
+    assert.deepStrictEqual(pubKeyMatch, true);
+  }
+}
+
+function validateSignature(
+  psbtParsed: ParsedPsbt2Of3 | ParsedPsbtP2TR | undefined,
+  txParsed: ParsedSignatureScript2Of3 | ParsedSignatureScriptTaproot | undefined
+) {
+  if (txParsed === undefined) {
+    assert.deepStrictEqual(psbtParsed?.signatures, undefined);
+  } else {
+    const txSignatures = txParsed.signatures.filter(
+      (txSig) => Buffer.isBuffer(txSig) && !isPlaceholderSignature(txSig)
+    );
+    assert.deepStrictEqual(txSignatures.length, psbtParsed?.signatures?.length);
+    if (txSignatures.length < 1) {
+      return;
+    }
+    const sigMatch = txSignatures.every((txSig) =>
+      Buffer.isBuffer(txSig) ? psbtParsed?.signatures?.some((psbtSig) => psbtSig.equals(txSig)) : true
+    );
+    assert.deepStrictEqual(sigMatch, true);
+  }
+}
+
+export function validatePsbtParsing(
+  tx: UtxoTransaction<bigint>,
+  psbt: UtxoPsbt<UtxoTransaction<bigint>>,
+  unspents: WalletUnspent<bigint>[],
+  signatureTarget: SignatureTargetType
+): void {
+  unspents.forEach((u, i) => {
+    if (isWalletUnspent(u)) {
+      const sciptType = scriptTypeForChain(u.chain);
+      const psbtParsed = parsePsbtInput(psbt, i);
+
+      if (signatureTarget === 'unsigned') {
+        if (sciptType === 'p2tr') {
+          assert.deepStrictEqual(psbtParsed, undefined);
+        } else {
+          assert.ok(psbtParsed);
+          assert.deepStrictEqual(psbtParsed?.scriptType, sciptType);
+          validateScript(psbtParsed, undefined);
+          validatePublicKeys(psbtParsed, undefined);
+          validateSignature(psbtParsed, undefined);
+        }
+      } else {
+        assert.ok(psbtParsed);
+        const txParsed = parseSignatureScript2Of3(tx.ins[i]);
+        assert.deepStrictEqual(psbtParsed?.scriptType, sciptType);
+        validateScript(psbtParsed, txParsed);
+        validatePublicKeys(psbtParsed, txParsed);
+        validateSignature(psbtParsed, txParsed);
+      }
+    }
+  });
+}
+
+export function toBigInt<TNumber extends number | bigint>(unspents: Unspent<TNumber>[]): WalletUnspent<bigint>[] {
+  return unspents.map((u) => {
+    if (isWalletUnspent(u)) {
+      return { ...u, value: BigInt(u.value) };
+    } else {
+      throw new Error(`invalid unspent`);
+    }
+  });
+}
+
+export function signPsbt(
+  psbt: UtxoPsbt<UtxoTransaction<bigint>>,
+  unspents: Unspent<bigint>[],
+  rootWalletKeys: RootWalletKeys,
+  signer: string,
+  cosigner: string,
+  signatureTarget: SignatureTargetType
+): void {
+  unspents.forEach((u, i) => {
+    if (isWalletUnspent(u)) {
```

Comment:
```suggestion
    if (!isWalletUnspent(u)) {
       throw new Error(`invalid unspent`);
    }
```

and then the main code can just have one fewer level of indentation. But you can instead just require `WalletUnspent` in the signature and move the check to the caller.

Reply:
Done